tests/testthat/test-correlation.R

```diff
@@ -36,6 +36,40 @@ test_that('correlation works with two data tables', {
 
 })
 
+test_that("correlation works with a single data table", {
+  nSamples = 200
+  testData1 <- data.table(
+    "contA1" = rnorm(nSamples),
+    "contB1" = rnorm(nSamples),
+    "contC1" = rnorm(nSamples)
+  )
+  corrResult <- correlation(testData1, method='spearman', verbose=F)
+  expect_equal(names(corrResult), c("data1","data2","correlationCoef"))
+  expect_equal(nrow(corrResult), (ncol(testData1) * ncol(testData1) - 3)/2)
```

Comment:
it's early so i migth not be functioning properly - why should we have (ncol*ncol-3)/2 rows?

Reply:
Make a square (ncol*ncol), remove the diagonal (-ncol, or 3), divide by two to dedup.

Reply:
ohhh thank you :) i missed the ncol=3 part